<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Query SignalP 5.0 web server. — get_signalp5 • ragp</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Query SignalP 5.0 web server. — get_signalp5"><meta property="og:description" content="The SignalP 5.0 server predicts the presence of signal peptides and the location of their cleavage sites in proteins from Archaea, Gram-positive Bacteria, Gram-negative Bacteria and Eukarya."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ragp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.5.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pkgdown/start.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/pkgdown/filter.html">Filtering hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/analyse.html">Analyse hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/PSV.html">Protein Sequence Visualization</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://ragp.shinyapps.io/Rapp/" class="external-link">ragp shiny</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/missuse/ragp/" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Query SignalP 5.0 web server.</h1>
    <small class="dont-index">Source: <a href="https://github.com/missuse/ragp/blob/HEAD/R/get_signalp5.R" class="external-link"><code>R/get_signalp5.R</code></a></small>
    <div class="hidden name"><code>get_signalp5.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The SignalP 5.0 server predicts the presence of signal peptides and the location of their cleavage sites in proteins from Archaea, Gram-positive Bacteria, Gram-negative Bacteria and Eukarya.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">get_signalp5</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for character</span>
<span class="fu">get_signalp5</span><span class="op">(</span>
  <span class="va">data</span>,
  org_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"euk"</span>, <span class="st">"gram-"</span>, <span class="st">"gram+"</span>, <span class="st">"archea"</span><span class="op">)</span>,
  splitter <span class="op">=</span> <span class="fl">2500L</span>,
  attempts <span class="op">=</span> <span class="fl">2</span>,
  progress <span class="op">=</span> <span class="cn">FALSE</span>,
  <span class="va">...</span>
<span class="op">)</span>

<span class="co"># S3 method for data.frame</span>
<span class="fu">get_signalp5</span><span class="op">(</span><span class="va">data</span>, <span class="va">sequence</span>, <span class="va">id</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for list</span>
<span class="fu">get_signalp5</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for default</span>
<span class="fu">get_signalp5</span><span class="op">(</span>data <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">sequence</span>, <span class="va">id</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for AAStringSet</span>
<span class="fu">get_signalp5</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://services.healthtech.dtu.dk/service.php?SignalP-5.0" class="external-link">https://services.healthtech.dtu.dk/service.php?SignalP-5.0</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data</dt>
<dd><p>A data frame with protein amino acid sequences as strings in one column and corresponding id's in another. Alternatively a path to a .fasta file with protein sequences. Alternatively a list with elements of class <code><a href="https://rdrr.io/pkg/seqinr/man/SeqFastaAA.html" class="external-link">SeqFastaAA</a></code> resulting from <code><a href="https://rdrr.io/pkg/seqinr/man/read.fasta.html" class="external-link">read.fasta</a></code> call. Alternatively an <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">AAStringSet</a></code> object. Should be left blank if vectors are provided to sequence and id arguments.</p></dd>
<dt>...</dt>
<dd><p>currently no additional arguments are accepted apart the ones documented bellow.</p></dd>
<dt>org_type</dt>
<dd><p>One of "euk", "gram-", "gram+" or "archea". Default is "euk". Are the protein sequences from Eukarya, Gram-negative Bacteria, Gram-positive Bacteria or Archaea.</p></dd>
<dt>splitter</dt>
<dd><p>An integer indicating the number of sequences to be in each .fasta file that is to be sent to the server. Default is 2500. Change only in case of a server side error. Accepted values are in range of 1 to 5000.</p></dd>
<dt>attempts</dt>
<dd><p>Integer, number of attempts if server unresponsive, at default set to 2.</p></dd>
<dt>progress</dt>
<dd><p>Boolean, whether to show messages of the job id for each batch. Default is FALSE.</p></dd>
<dt>sequence</dt>
<dd><p>A vector of strings representing protein amino acid sequences, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></dd>
<dt>id</dt>
<dd><p>A vector of strings representing protein identifiers, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>if org_type is "euk" a data frame with columns:</p><dl><dt>id</dt>
<dd><p>Character, as from input</p></dd>

  <dt>Prediction</dt>
<dd><p>Integer, The type of signal peptide predicted: Sec/SPI, Tat/SPI, Sec/SPII or Other if no signal peptide predicted</p></dd>

  <dt>SP.Sec.SPI</dt>
<dd><p>Numeric, marginal probability that the protein contains a Sec  N-terminal signal peptide (Sec/SPI).</p></dd>

  <dt>Other</dt>
<dd><p>Numeric, the probability that the sequence does not have any kind of signal peptid.</p></dd>

  <dt>CS_pos</dt>
<dd><p>Character, cleavage site position.</p></dd>

  <dt>Pr</dt>
<dd><p>Numeric, probability of the predicted cleavage site position.</p></dd>

  <dt>cleave.site</dt>
<dd><p>Numeric,llocal amino acid sequence arround the predicted cleavage site.</p></dd>

  <dt>is.signalp</dt>
<dd><p>Logical, did SignalP5 predict the presence of a signal peptide.</p></dd>

  <dt>sp.length</dt>
<dd><p>Integer, length of the predicted signal peptide.</p></dd>

  
</dl><p>if org_type is one of "gram-", "gram+" or "archea" the returned data frame will have two additional columns between `SP.Sec.SPI` and `Other`:</p><dl><dt>TAT.Tat.SPI</dt>
<dd><p>Numeric, marginal probability that the protein contains a Tat N-terminal signal peptide (Tat/SPI).</p></dd>

  <dt>LIPO.Sec.SPII</dt>
<dd><p>Numeric, marginal probability that the protein contains a Lipoprotein N-terminal signal peptide (Sec/SPII).</p></dd>

  
</dl></div>
    <div id="note">
    <h2>Note</h2>
    <p>This function creates temporary files in the working directory. If something goes wrong during communication with the server and progress was set to TRUE, predictions can be obtained using `file.path("http://www.cbs.dtu.dk/services/SignalP-5.0/tmp", jobid, "output_protein_type.txt")` eg `read.delim(file.path(...), header = TRUE, skip = 1)`.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Almagro Armenteros JJ, Tsirigos KD, Sønderby CK, Petersen TN, Winther O, Brunak S, von Heijne G, Nielsen H. (2019) SignalP 5.0 improves signal peptide predictions using deep neural networks. Nature Biotechnology, 37:420-423, doi:10.1038/s41587-019-0036-z</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="get_signalp.html">get_signalp</a></code> <code><a href="get_targetp.html">get_targetp</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/missuse/ragp" class="external-link">ragp</a></span><span class="op">)</span></span>
<span class="r-in"><span class="va">sp5_pred</span> <span class="op">&lt;-</span> <span class="fu">get_signalp5</span><span class="op">(</span>data <span class="op">=</span> <span class="va">at_nsp</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span>,</span>
<span class="r-in">                         <span class="va">sequence</span>,</span>
<span class="r-in">                         <span class="va">Transcript.id</span><span class="op">)</span></span>
<span class="r-in"><span class="va">sp5_pred</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             id  Prediction SP.Sec.SPI    Other CS_pos     Pr cleave.site</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  ATCG00660.1       OTHER   0.000375 0.999625            NA            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  AT2G43600.1 SP(Sec/SPI)   0.999802 0.000198  22-23 0.9639      VFS-QN</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  AT2G28410.1 SP(Sec/SPI)   0.990424 0.009576  22-23 0.8897      ALA-QD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  AT2G22960.1 SP(Sec/SPI)   0.998142 0.001858  22-23 0.9424      AES-GS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  AT2G19580.1       OTHER   0.264792 0.735208            NA            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  AT2G19690.2 SP(Sec/SPI)   0.989540 0.010460  28-29 0.9030      ARS-EE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  AT2G19690.1 SP(Sec/SPI)   0.989540 0.010460  28-29 0.9030      ARS-EE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  AT2G33130.1 SP(Sec/SPI)   0.959278 0.040722  26-27 0.5119      VVG-SR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  AT2G05520.1 SP(Sec/SPI)   0.999493 0.000507  23-24 0.5666      VAA-AS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 AT2G05520.2 SP(Sec/SPI)   0.999493 0.000507  23-24 0.5666      VAA-AS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    is.signalp sp.length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       FALSE        NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2        TRUE        22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3        TRUE        22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4        TRUE        22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5       FALSE        NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6        TRUE        28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7        TRUE        28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8        TRUE        26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9        TRUE        23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10       TRUE        23</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Milan Dragicevic, Danijela Paunovic.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

