<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Query Espritz web server. — get_espritz • ragp</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Query Espritz web server. — get_espritz" />
<meta property="og:description" content="Espritz web server predicts disordered regions from primary sequence. It utilizes Bi-directional Recursive Neural Networks and can process proteins on a genomic scale with little effort and state-of-the-art accuracy." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ragp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.5.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pkgdown/start.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/pkgdown/filter.html">Filtering hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/analyse.html">Analyse hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/PSV.html">Protein Sequence Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://ragp.shinyapps.io/Rapp/">ragp shiny</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/missuse/ragp/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Query Espritz web server.</h1>
    <small class="dont-index">Source: <a href='https://github.com/missuse/ragp/blob/master/R/get_espritz.R'><code>R/get_espritz.R</code></a></small>
    <div class="hidden name"><code>get_espritz.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Espritz web server predicts disordered regions from primary sequence. It utilizes Bi-directional Recursive Neural Networks and can process proteins on a genomic scale with little effort and state-of-the-art accuracy.</p>
    </div>

    <pre class="usage"><span class='fu'>get_espritz</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for character</span>
<span class='fu'>get_espritz</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for data.frame</span>
<span class='fu'>get_espritz</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>sequence</span>, <span class='va'>id</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for list</span>
<span class='fu'>get_espritz</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for default</span>
<span class='fu'>get_espritz</span><span class='op'>(</span>
  data <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>sequence</span>,
  <span class='va'>id</span>,
  model <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"X-Ray"</span>, <span class='st'>"Disprot"</span>, <span class='st'>"NMR"</span><span class='op'>)</span>,
  FPR <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"best Sw"</span>, <span class='st'>"5% FPR"</span><span class='op'>)</span>,
  simplify <span class='op'>=</span> <span class='cn'>TRUE</span>,
  progress <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S3 method for AAStringSet</span>
<span class='fu'>get_espritz</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>A data frame with protein amino acid sequences as strings in one column and corresponding id's in another. Alternatively a path to a .fasta file with protein sequences. Alternatively a list with elements of class <code><a href='https://rdrr.io/pkg/seqinr/man/SeqFastaAA.html'>SeqFastaAA</a></code> resulting from <code><a href='https://rdrr.io/pkg/seqinr/man/read.fasta.html'>read.fasta</a></code> call. Alternatively an <code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>AAStringSet</a></code> object. Should be left blank if vectors are provided to sequence and id arguments.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>currently no additional arguments are accepted apart the ones documented bellow.</p></td>
    </tr>
    <tr>
      <th>sequence</th>
      <td><p>A vector of strings representing protein amino acid sequences, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    <tr>
      <th>id</th>
      <td><p>A vector of strings representing protein identifiers, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    <tr>
      <th>model</th>
      <td><p>One of c('X-Ray', 'Disprot', 'NMR'), default is 'X-Ray'. Determines the model to be used for prediction. See details.</p></td>
    </tr>
    <tr>
      <th>FPR</th>
      <td><p>One of c('best Sw', '5"%" FPR'). default is 'best Sw'. Determines the cutoff probability for prediction. 'best Sw' maximizes a weighted score rewarding correctly disorder prediction more than order prediction.</p></td>
    </tr>
    <tr>
      <th>simplify</th>
      <td><p>A Boolean indicating the type of returned object, defaults to TRUE.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p>Boolean, whether to show the progress bar, at default set to FALSE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p><a href='http://old.protein.bio.unipd.it/espritz/'>http://old.protein.bio.unipd.it/espritz/</a></p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>If simplify == TRUE:
A data frame (one row per disordered region) with columns:</p><dl>
  <dt>start</dt><dd><p>Integer, indicating the sequence position of disordered region start.</p></dd>
  <dt>end</dt><dd><p>Integer, indicating the sequence position of disordered region end.</p></dd>
  <dt>id</dt><dd><p>Character, indicating the protein identifier.</p></dd>
  
</dl>

<p>If simplify == FALSE:
A data frame (one row per protein) with columns:</p><dl>
  <dt>id</dt><dd><p>Character, indicating the protein identifier.</p></dd>
  <dt>probability</dt><dd><p>List column of numeric vectors, vectors contain probabilities of disorder for each residue.</p></dd>
  <dt>prediction</dt><dd><p>Character, indicating the prediction: D - disordered, O - ordered for each residue.</p></dd>
  
</dl>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Three models trained on different data sets are available and can be selected via the argument model: X-Ray - based on missing atoms from the Protein Data Bank (PDB) X-ray solved structures. If this option is chosen then the predictors with short disorder options are executed. Disprot -  contains longer disorder segments compared to x-ray. In particular, disprot a manually curetted database which is often based on functional attributes of the disordered region was used for this definition. Disorder residues are defined if the disprot curators consider the residue to be disordered at least once. All other residues are considered ordered. If this option is chosen then the predictors with long disorder options are executed. 'NMR' - based on NMR mobility. NMR flexibility is calculated using the Mobi server optimized to replicate the ordered-disordered NMR definition used in CASP8. These models provide quite different predictions. For further details visit <a href='http://old.protein.bio.unipd.it/espritz/help_pages/help.html'>http://old.protein.bio.unipd.it/espritz/help_pages/help.html</a> and <a href='http://old.protein.bio.unipd.it/espritz/help_pages/methods.html'>http://old.protein.bio.unipd.it/espritz/help_pages/methods.html</a>.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>The Espritz web server has a limit on the amount of daily queries by ip. The function will inform the user when the limit has been exceeded.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Walsh I, Martin AJM, Di domenico T, Tosatto SCE (2012) ESpritz: accurate and fast prediction of protein disorder. Bioinformatics 28(4): 503 - 509</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/missuse/ragp'>ragp</a></span><span class='op'>)</span>

<span class='va'>espritz_test</span> <span class='op'>&lt;-</span> <span class='fu'>get_espritz</span><span class='op'>(</span><span class='va'>at_nsp</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>10</span>,<span class='op'>]</span>,
                            <span class='va'>sequence</span>,
                            <span class='va'>Transcript.id</span><span class='op'>)</span>
<span class='va'>espritz_test</span>
</div><div class='output co'>#&gt;    start end          id
#&gt; 1      1  12 ATCG00660.1
#&gt; 2     49  54 ATCG00660.1
#&gt; 3    111 117 ATCG00660.1
#&gt; 4      1   9 AT2G43600.1
#&gt; 5    266 273 AT2G43600.1
#&gt; 6      1   8 AT2G28410.1
#&gt; 7     47  61 AT2G28410.1
#&gt; 8    115 115 AT2G28410.1
#&gt; 9      1   8 AT2G22960.1
#&gt; 10    16  19 AT2G22960.1
#&gt; 11   178 184 AT2G22960.1
#&gt; 12     1   9 AT2G19580.1
#&gt; 13    83  88 AT2G19580.1
#&gt; 14   256 270 AT2G19580.1
#&gt; 15     1  12 AT2G19690.2
#&gt; 16   104 108 AT2G19690.2
#&gt; 17   143 148 AT2G19690.2
#&gt; 18     1  12 AT2G19690.1
#&gt; 19   104 109 AT2G19690.1
#&gt; 20   139 147 AT2G19690.1
#&gt; 21     1  10 AT2G33130.1
#&gt; 22    15  17 AT2G33130.1
#&gt; 23    65 103 AT2G33130.1
#&gt; 24     1 128 AT2G05520.1
#&gt; 25   130 145 AT2G05520.1
#&gt; 26     1 118 AT2G05520.2
#&gt; 27   123 138 AT2G05520.2</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Milan Dragicevic, Danijela Paunovic.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


