<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Query SignalP web server. — get_signalp • ragp</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Query SignalP web server. — get_signalp" />
<meta property="og:description" content="SignalP 4.1 server predicts the presence and location of signal peptide cleavage sites in amino acid sequences from different organisms: Gram-positive prokaryotes, Gram-negative prokaryotes, and eukaryotes. The method incorporates a prediction of cleavage sites and a signal peptide/non-signal peptide prediction based on a combination of several artificial neural networks." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ragp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.5.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pkgdown/start.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/pkgdown/filter.html">Filtering hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/analyse.html">Analyse hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/PSV.html">Protein Sequence Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://ragp.shinyapps.io/Rapp/">ragp shiny</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/missuse/ragp/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Query SignalP web server.</h1>
    <small class="dont-index">Source: <a href='https://github.com/missuse/ragp/blob/master/R/get_signalp.R'><code>R/get_signalp.R</code></a></small>
    <div class="hidden name"><code>get_signalp.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>SignalP 4.1 server predicts the presence and location of signal peptide cleavage sites in amino acid sequences from different organisms: Gram-positive prokaryotes, Gram-negative prokaryotes, and eukaryotes. The method incorporates a prediction of cleavage sites and a signal peptide/non-signal peptide prediction based on a combination of several artificial neural networks.</p>
    </div>

    <pre class="usage"><span class='fu'>get_signalp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for character</span>
<span class='fu'>get_signalp</span><span class='op'>(</span>
  <span class='va'>data</span>,
  org_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"euk"</span>, <span class='st'>"gram-"</span>, <span class='st'>"gram+"</span><span class='op'>)</span>,
  Dcut_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"default"</span>, <span class='st'>"sensitive"</span>, <span class='st'>"user"</span><span class='op'>)</span>,
  Dcut_noTM <span class='op'>=</span> <span class='fl'>0.45</span>,
  Dcut_TM <span class='op'>=</span> <span class='fl'>0.5</span>,
  method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"best"</span>, <span class='st'>"notm"</span><span class='op'>)</span>,
  minlen <span class='op'>=</span> <span class='cn'>NULL</span>,
  trunc <span class='op'>=</span> <span class='fl'>70L</span>,
  splitter <span class='op'>=</span> <span class='fl'>1000L</span>,
  attempts <span class='op'>=</span> <span class='fl'>2</span>,
  progress <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S3 method for data.frame</span>
<span class='fu'>get_signalp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>sequence</span>, <span class='va'>id</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for list</span>
<span class='fu'>get_signalp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for default</span>
<span class='fu'>get_signalp</span><span class='op'>(</span>data <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>sequence</span>, <span class='va'>id</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for AAStringSet</span>
<span class='fu'>get_signalp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>A data frame with protein amino acid sequences as strings in one column and corresponding id's in another. Alternatively a path to a .fasta file with protein sequences. Alternatively a list with elements of class <code><a href='https://rdrr.io/pkg/seqinr/man/SeqFastaAA.html'>SeqFastaAA</a></code> resulting from <code><a href='https://rdrr.io/pkg/seqinr/man/read.fasta.html'>read.fasta</a></code> call. Alternatively an <code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>AAStringSet</a></code> object. Should be left blank if vectors are provided to sequence and id arguments.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>currently no additional arguments are accepted apart the ones documented bellow.</p></td>
    </tr>
    <tr>
      <th>org_type</th>
      <td><p>One of c("euk", "gram-", "gram+"), defaults to "euk". Which model should be used for prediction.</p></td>
    </tr>
    <tr>
      <th>Dcut_type</th>
      <td><p>One of c("default", "sensitive", "user"), defaults to "default". The default cutoff values for SignalP 4 are chosen to optimize the performance measured as Matthews Correlation Coefficient (MCC). This results in a lower sensitivity (true positive rate) than SignalP 3.0 had. Setting this argument to "sensitive" will yield the same sensitivity as SignalP 3.0. This will make the false positive rate slightly higher, but still better than that of SignalP 3.0.</p></td>
    </tr>
    <tr>
      <th>Dcut_noTM</th>
      <td><p>A numeric value, with range 0 - 1, defaults to 0.45. For experimenting with cutoff values.</p></td>
    </tr>
    <tr>
      <th>Dcut_TM</th>
      <td><p>A numeric value, with range 0 - 1, defaults to 0.5. For experimenting with cutoff values.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>One of c("best", "notm"), defaults to "best". Signalp 4.1 contains two types of neural networks. SignalP-TM has been trained with sequences containing transmembrane segments in the data set, while SignalP-noTM has been trained without those sequences. Per default, SignalP 4.1 uses SignalP-TM as a preprocessor to determine whether to use SignalP-TM or SignalP-noTM in the final prediction (if 4 or more positions are predicted to be in a transmembrane state, SignalP-TM is used, otherwise SignalP-noTM). An exception is Gram-positive bacteria, where SignalP-TM is used always. If you are confident that there are no transmembrane segments in your data, you can get a slightly better performance by choosing "Input sequences do not include TM regions", which will tell SignalP 4.1 to use SignalP-noTM always.</p></td>
    </tr>
    <tr>
      <th>minlen</th>
      <td><p>An integer value corresponding to the minimal predicted signal peptide length, at default set to 10. SignalP 4.0 could, in rare cases, erroneously predict signal peptides shorter than 10 residues. These errors have in SignalP 4.1 been eliminated by imposing a lower limit on the cleavage site position (signal peptide length). The minimum length is by default 10, but you can adjust it. Signal peptides shorter than 15 residues are very rare. If you want to disable this length restriction completely, enter 0 (zero).</p></td>
    </tr>
    <tr>
      <th>trunc</th>
      <td><p>An integer value corresponding to the N-terminal truncation of input sequence, at default set to 70. By default, the predictor truncates each sequence to max. 70 residues before submitting it to the neural networks. If you want to predict extremely long signal peptides, you can try a higher value, or disable truncation completely by entering 0 (zero).</p></td>
    </tr>
    <tr>
      <th>splitter</th>
      <td><p>An integer indicating the number of sequences to be in each .fasta file that is to be sent to the server. Default is 1000. Change only in case of a server side error. Accepted values are in range of 1 to 2000.</p></td>
    </tr>
    <tr>
      <th>attempts</th>
      <td><p>Integer, number of attempts if server unresponsive, at default set to 2.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p>Boolean, whether to show the progress bar, at default set to FALSE.</p></td>
    </tr>
    <tr>
      <th>sequence</th>
      <td><p>A vector of strings representing protein amino acid sequences, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    <tr>
      <th>id</th>
      <td><p>A vector of strings representing protein identifiers, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p><a href='https://services.healthtech.dtu.dk/service.php?SignalP-4.1'>https://services.healthtech.dtu.dk/service.php?SignalP-4.1</a></p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data frame with columns:</p><dl>
  <dt>id</dt><dd><p>Character, as from input</p></dd>
  <dt>Cmax</dt><dd><p>Numeric, C-score (raw cleavage site score). The output from the CS networks, which are trained to distinguish signal peptide cleavage sites from everything else. Note the position numbering of the cleavage site: the C-score is trained to be high at the position immediately after the cleavage site (the first residue in the mature protein).</p></dd>
  <dt>Cmax.pos</dt><dd><p>Integer, position of Cmax. position immediately after the cleavage site (the first residue in the mature protein).</p></dd>
  <dt>Ymax</dt><dd><p>Numeric, Y-score (combined cleavage site score), A combination (geometric average) of the C-score and the slope of the S-score, resulting in a better cleavage site prediction than the raw C-score alone. This is due to the fact that multiple high-peaking C-scores can be found in one sequence, where only one is the true cleavage site. The Y-score distinguishes between C-score peaks by choosing the one where the slope of the S-score is steep.</p></dd>
  <dt>Ymax.pos</dt><dd><p>Integer, position of Ymax</p></dd>
  <dt>Smax</dt><dd><p>Numeric, S-score (signal peptide score). The output from the SP networks, which are trained to distinguish positions within signal peptides from positions in the mature part of the proteins and from proteins without signal peptides.</p></dd>
  <dt>Smax.pos</dt><dd><p>Integer, position of Smax</p></dd>
  <dt>Smean</dt><dd><p>Numeric, The average S-score of the possible signal peptide (from position 1 to the position immediately before the maximal Y-score)</p></dd>
  <dt>Dmean</dt><dd><p>Numeric, D-score (discrimination score). A weighted average of the mean S and the max. Y scores. This is the score that is used to discriminate signal peptides from non-signal peptides.</p></dd>
  <dt>is.sp</dt><dd><p>Character, does the sequence contain a N-sp</p></dd>
  <dt>Dmaxcut</dt><dd><p>Numeric, as from input, Dcut_noTM if SignalP-noTM network used and Dcut_TM if SignalP-TM network used</p></dd>
  <dt>Networks.used</dt><dd><p>Character, which network was used for the prediction: SignalP-noTM or SignalP-TM</p></dd>
  <dt>is.signalp</dt><dd><p>Logical, did SignalP predict the presence of a signal peptide</p></dd>
  <dt>sp.length</dt><dd><p>Integer, length of the predicted signal peptide.</p></dd>
  
</dl>

    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>This function creates temporary files in the working directory.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Petersen TN. Brunak S. Heijne G. Nielsen H. (2011) SignalP 4.0: discriminating signal peptides from transmembrane regions. Nature Methods 8: 785-786</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='get_signalp5.html'>get_signalp5</a></code> <code><a href='get_phobius.html'>get_phobius</a></code> <code><a href='get_targetp.html'>get_targetp</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/missuse/ragp'>ragp</a></span><span class='op'>)</span>
<span class='va'>signalp_pred</span> <span class='op'>&lt;-</span> <span class='fu'>get_signalp</span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>at_nsp</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>10</span>,<span class='op'>]</span>,
                            <span class='va'>sequence</span>,
                            <span class='va'>Transcript.id</span><span class='op'>)</span>
<span class='va'>signalp_pred</span>
</div><div class='output co'>#&gt;             id  Cmax Cmax.pos  Ymax Ymax.pos  Smax Smax.pos Smean Dmean is.sp
#&gt; 1  ATCG00660.1 0.210       30 0.162       30 0.260        2 0.146 0.154     N
#&gt; 2  AT2G43600.1 0.860       23 0.894       23 0.984       13 0.930 0.913     Y
#&gt; 3  AT2G28410.1 0.779       23 0.826       23 0.940       15 0.877 0.853     Y
#&gt; 4  AT2G22960.1 0.701       23 0.790       23 0.948       15 0.891 0.844     Y
#&gt; 5  AT2G19580.1 0.422       26 0.586       26 0.885       17 0.798 0.671     Y
#&gt; 6  AT2G19690.2 0.797       29 0.870       29 0.987       18 0.952 0.914     Y
#&gt; 7  AT2G19690.1 0.797       29 0.870       29 0.987       18 0.952 0.914     Y
#&gt; 8  AT2G33130.1 0.318       33 0.530       27 0.990       16 0.933 0.748     Y
#&gt; 9  AT2G05520.1 0.633       24 0.782       24 0.990       16 0.966 0.881     Y
#&gt; 10 AT2G05520.2 0.633       24 0.782       24 0.990       16 0.966 0.881     Y
#&gt;    Dmaxcut Networks.used is.signalp sp.length
#&gt; 1    0.450  SignalP-noTM      FALSE        30
#&gt; 2    0.450  SignalP-noTM       TRUE        23
#&gt; 3    0.450  SignalP-noTM       TRUE        23
#&gt; 4    0.450  SignalP-noTM       TRUE        23
#&gt; 5    0.500    SignalP-TM       TRUE        26
#&gt; 6    0.450  SignalP-noTM       TRUE        29
#&gt; 7    0.450  SignalP-noTM       TRUE        29
#&gt; 8    0.450  SignalP-noTM       TRUE        27
#&gt; 9    0.450  SignalP-noTM       TRUE        24
#&gt; 10   0.450  SignalP-noTM       TRUE        24</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Milan Dragicevic, Danijela Paunovic.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


