<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Query TargetP web server. — get_targetp • ragp</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Query TargetP web server. — get_targetp" />
<meta property="og:description" content="TargetP 1.1 predicts the subcellular location of eukaryotic proteins. The location assignment is based on the predicted presence of any of the N-terminal presequences: chloroplast transit peptide (cTP), mitochondrial targeting peptide (mTP) or secretory pathway signal peptide (SP). TargetP uses ChloroP and SignalP to predict cleavage sites for cTP and SP, respectively. For the sequences predicted to contain an N-terminal presequence a potential cleavage site is also predicted." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ragp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pkgdown/start.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/pkgdown/filter.html">Filtering hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/analyse.html">Analyse hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/PSV.html">Protein Sequence Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://ragp.shinyapps.io/Rapp/">ragp shiny</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/missuse/ragp/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Query TargetP web server.</h1>
    <small class="dont-index">Source: <a href='https://github.com/missuse/ragp/blob/master/R/get_targetp.R'><code>R/get_targetp.R</code></a></small>
    <div class="hidden name"><code>get_targetp.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>TargetP 1.1 predicts the subcellular location of eukaryotic proteins. The location assignment is based on the predicted presence of any of the N-terminal presequences: chloroplast transit peptide (cTP), mitochondrial targeting peptide (mTP) or secretory pathway signal peptide (SP). TargetP uses ChloroP and SignalP to predict cleavage sites for cTP and SP, respectively. For the sequences predicted to contain an N-terminal presequence a potential cleavage site is also predicted.</p>
    </div>

    <pre class="usage"><span class='fu'>get_targetp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for character</span>
<span class='fu'>get_targetp</span><span class='op'>(</span>
  <span class='va'>data</span>,
  org_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"non_plant"</span>, <span class='st'>"plant"</span><span class='op'>)</span>,
  cutoffs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"winner_takes_all"</span>, <span class='st'>"spec95"</span>, <span class='st'>"spec90"</span>, <span class='st'>"custom"</span><span class='op'>)</span>,
  tcut <span class='op'>=</span> <span class='cn'>NULL</span>,
  pcut <span class='op'>=</span> <span class='cn'>NULL</span>,
  scut <span class='op'>=</span> <span class='cn'>NULL</span>,
  ocut <span class='op'>=</span> <span class='cn'>NULL</span>,
  splitter <span class='op'>=</span> <span class='fl'>200</span>,
  sleep <span class='op'>=</span> <span class='fl'>3</span>,
  attempts <span class='op'>=</span> <span class='fl'>2</span>,
  progress <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S3 method for data.frame</span>
<span class='fu'>get_targetp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>sequence</span>, <span class='va'>id</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for list</span>
<span class='fu'>get_targetp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for default</span>
<span class='fu'>get_targetp</span><span class='op'>(</span>data <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>sequence</span>, <span class='va'>id</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for AAStringSet</span>
<span class='fu'>get_targetp</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>A data frame with protein amino acid sequences as strings in one column and corresponding id's in another. Alternatively a path to a .fasta file with protein sequences. Alternatively a list with elements of class <code><a href='https://rdrr.io/pkg/seqinr/man/SeqFastaAA.html'>SeqFastaAA</a></code> resulting from <code><a href='https://rdrr.io/pkg/seqinr/man/read.fasta.html'>read.fasta</a></code> call. Alternatively an <code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>AAStringSet</a></code> object. Should be left blank if vectors are provided to sequence and id arguments.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>currently no additional arguments are accepted apart the ones documented bellow.</p></td>
    </tr>
    <tr>
      <th>org_type</th>
      <td><p>One of c("non_plant", "plant"), defaults to "plant". Which models should be used for prediction.</p></td>
    </tr>
    <tr>
      <th>cutoffs</th>
      <td><p>One of c("winner_takes_all", "spec95", "spec90", "custom"), defaults to "winner_takes_all". If "winner_takes_all" no cutoffs are specified, if "spec95" or "spec90" are selected, predefined set of cutoffs that yielded &gt;0.95 or &gt;0.9 specificity on the TargetP test sets. If "custom" specified user defined cutoffs should be specified in "tcut", "pcut", "scut", ocut".</p></td>
    </tr>
    <tr>
      <th>tcut</th>
      <td><p>A numeric value, with range 0 - 1, defaults to 0 (cutoff = "winner_takes_all"). mTP user specified cutoff.</p></td>
    </tr>
    <tr>
      <th>pcut</th>
      <td><p>A numeric value, with range 0 - 1, defaults to 0 (cutoff = "winner_takes_all"). cTP user specified cutoff.</p></td>
    </tr>
    <tr>
      <th>scut</th>
      <td><p>A numeric value, with range 0 - 1, defaults to 0 (cutoff = "winner_takes_all"). SP user specified cutoff.</p></td>
    </tr>
    <tr>
      <th>ocut</th>
      <td><p>A numeric value, with range 0 - 1, defaults to 0 (cutoff = "winner_takes_all"). User specified cutoff for "other" (not with mTP, cTP, SP).</p></td>
    </tr>
    <tr>
      <th>splitter</th>
      <td><p>An integer indicating the number of sequences to be in each .fasta file that is to be sent to the server. Defaults to 200. Change only in case of a server side error. Accepted values are in range of 1 to 2000.</p></td>
    </tr>
    <tr>
      <th>sleep</th>
      <td><p>A numeric indicating the pause in seconds between server calls, at default set to 1</p></td>
    </tr>
    <tr>
      <th>attempts</th>
      <td><p>Integer, number of attempts if server unresponsive, at default set to 2.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p>Boolean, whether to show the progress bar, at default set to FALSE.</p></td>
    </tr>
    <tr>
      <th>sequence</th>
      <td><p>A vector of strings representing protein amino acid sequences, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    <tr>
      <th>id</th>
      <td><p>A vector of strings representing protein identifiers, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p><a href='https://services.healthtech.dtu.dk/service.php?TargetP-1.1'>https://services.healthtech.dtu.dk/service.php?TargetP-1.1</a></p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data frame with columns:</p><dl>
  <dt>Name</dt><dd><p>Character, sequence name truncated to 20 characters</p></dd>
  <dt>Len</dt><dd><p>Integer, length of analyzed sequence</p></dd>
  <dt>cTP</dt><dd><p>Numeric, final NN sequence score to contain a chloroplast transit peptide (cTP). ChloroP is used to predict cleavage sites for cTP</p></dd>
  <dt>mTP</dt><dd><p>Numeric, final NN sequence score to contain a mitochondrial targeting peptide (mTP).</p></dd>
  <dt>SP</dt><dd><p>Numeric, final NN sequence score to contain a secretory pathway signal peptide (SP). SignalP is used to predict cleavage sites for SP</p></dd>
  <dt>other</dt><dd><p>Numeric, final NN sequence score of a sequence not to contain mTP, SP or cTP</p></dd>
  <dt>Loc</dt><dd><p>Character, one of C (chloroplast), M (mitochondrion), S (secretory pathway), - (any other location) or * ("don't know"; indicates that cutoff restrictions were set and the winning network output score was below the requested cutoff for that category.)</p></dd>
  <dt>TPlen</dt><dd><p>Integer, predicted presequence length</p></dd>
  <dt>RC</dt><dd><p>Integer, reliability class, from 1 to 5, where 1 indicates the strongest prediction. RC is a measure of the size of the difference ('diff') between the highest (winning) and the second highest output scores. There are 5 reliability classes, defined as follows: 1 : diff &gt; 0.800, 2 : 0.800 &gt; diff &gt; 0.600, 3 : 0.600 &gt; diff &gt; 0.400, 4 : 0.400 &gt; diff &gt; 0.200, 5 : 0.200 &gt; diff. Thus, the lower the value of RC the safer the prediction.</p></dd>
  <dt>is.targetp</dt><dd><p>Logical, did TargetP predict the presence of a signal peptide</p></dd>
  
</dl>

    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>This function creates temporary files in the working directory. Protein ids should be shorter then 20 characters due to server side truncation.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Emanuelsson O, Nielsen H, Brunak S,von Heijne G. (2000) Predicting subcellular localization of proteins based on their N-terminal amino acid sequence. J. Mol. Biol.300: 1005-1016</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='get_signalp.html'>get_signalp</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/missuse/ragp'>ragp</a></span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>at_nsp</span><span class='op'>)</span>
<span class='va'>targetp_pred</span> <span class='op'>&lt;-</span> <span class='fu'>get_targetp</span><span class='op'>(</span><span class='va'>at_nsp</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>20</span>,<span class='op'>]</span>,
                            <span class='va'>sequence</span>,
                            <span class='va'>Transcript.id</span><span class='op'>)</span>
<span class='va'>targetp_pred</span>     
</div><div class='output co'>#&gt;           Name Len   cTP   mTP    SP other Loc RC TPlen is.targetp
#&gt; 1  ATCG00660.1 117 0.219 0.741 0.013 0.084   M  3    20      FALSE
#&gt; 2  AT2G43600.1 273 0.031 0.014 0.991 0.049   S  1    22       TRUE
#&gt; 3  AT2G28410.1 115 0.028 0.027 0.976 0.011   S  1    22       TRUE
#&gt; 4  AT2G22960.1 184 0.011 0.109 0.941 0.025   S  1    22       TRUE
#&gt; 5  AT2G19580.1 270 0.003 0.053 0.868 0.079   S  2    25       TRUE
#&gt; 6  AT2G19690.2 148 0.002 0.227 0.987 0.007   S  2    28       TRUE
#&gt; 7  AT2G19690.1 147 0.002 0.227 0.987 0.007   S  2    28       TRUE
#&gt; 8  AT2G33130.1 103 0.011 0.017 0.993 0.037   S  1    32       TRUE
#&gt; 9  AT2G05520.1 145 0.008 0.016 0.997 0.039   S  1    23       TRUE
#&gt; 10 AT2G05520.2 138 0.008 0.018 0.998 0.036   S  1    23       TRUE
#&gt; 11 AT2G05520.3 125 0.009 0.013 0.997 0.041   S  1    23       TRUE
#&gt; 12 AT2G05520.4 124 0.009 0.012 0.997 0.044   S  1    23       TRUE
#&gt; 13 AT2G05520.6 117 0.010 0.012 0.998 0.038   S  1    23       TRUE
#&gt; 14 AT2G05520.5 111 0.009 0.017 0.998 0.027   S  1    23       TRUE
#&gt; 15 AT2G04031.1 151 0.033 0.042 0.944 0.053   S  1    24       TRUE
#&gt; 16 AT2G43620.1 283 0.002 0.182 0.966 0.008   S  2    28       TRUE
#&gt; 17 AT2G37390.1 259 0.103 0.276 0.156 0.404   _  5     -      FALSE
#&gt; 18 AT2G37390.2 258 0.103 0.276 0.156 0.404   _  5     -      FALSE
#&gt; 19 AT2G35250.1 342 0.010 0.267 0.784 0.153   S  3    21       TRUE
#&gt; 20 AT2G38390.1 349 0.126 0.055 0.899 0.014   S  2    29       TRUE</div><div class='input'>                      
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Milan Dragicevic, Danijela Paunovic.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


