<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Query SignalP 5.0 web server. — get_signalp5 • ragp</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Query SignalP 5.0 web server. — get_signalp5" />
<meta property="og:description" content="The SignalP 5.0 server predicts the presence of signal peptides and the location of their cleavage sites in proteins from Archaea, Gram-positive Bacteria, Gram-negative Bacteria and Eukarya." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ragp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pkgdown/start.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/pkgdown/filter.html">Filtering hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/analyse.html">Analyse hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/PSV.html">Protein Sequence Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://ragp.shinyapps.io/Rapp/">ragp shiny</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/missuse/ragp/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Query SignalP 5.0 web server.</h1>
    <small class="dont-index">Source: <a href='https://github.com/missuse/ragp/blob/master/R/get_signalp5.R'><code>R/get_signalp5.R</code></a></small>
    <div class="hidden name"><code>get_signalp5.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The SignalP 5.0 server predicts the presence of signal peptides and the location of their cleavage sites in proteins from Archaea, Gram-positive Bacteria, Gram-negative Bacteria and Eukarya.</p>
    </div>

    <pre class="usage"><span class='fu'>get_signalp5</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for character</span>
<span class='fu'>get_signalp5</span><span class='op'>(</span>
  <span class='va'>data</span>,
  org_type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"euk"</span>, <span class='st'>"gram-"</span>, <span class='st'>"gram+"</span>, <span class='st'>"archea"</span><span class='op'>)</span>,
  splitter <span class='op'>=</span> <span class='fl'>2500L</span>,
  attempts <span class='op'>=</span> <span class='fl'>2</span>,
  progress <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S3 method for data.frame</span>
<span class='fu'>get_signalp5</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>sequence</span>, <span class='va'>id</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for list</span>
<span class='fu'>get_signalp5</span><span class='op'>(</span><span class='va'>data</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S3 method for default</span>
<span class='fu'>get_signalp5</span><span class='op'>(</span>data <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>sequence</span>, <span class='va'>id</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p>A data frame with protein amino acid sequences as strings in one column and corresponding id's in another. Alternatively a path to a .fasta file with protein sequences. Alternatively a list with elements of class "SeqFastaAA" resulting from <code><a href='https://rdrr.io/pkg/seqinr/man/read.fasta.html'>read.fasta</a></code> call. Should be left blank if vectors are provided to sequence and id arguments.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>currently no additional arguments are accepted apart the ones documented bellow.</p></td>
    </tr>
    <tr>
      <th>org_type</th>
      <td><p>One of "euk", "gram-", "gram+" or "archea". Default is "euk". Are the protein sequences from Eukarya, Gram-negative Bacteria, Gram-positive Bacteria or Archaea.</p></td>
    </tr>
    <tr>
      <th>splitter</th>
      <td><p>An integer indicating the number of sequences to be in each .fasta file that is to be sent to the server. Default is 2500. Change only in case of a server side error. Accepted values are in range of 1 to 5000.</p></td>
    </tr>
    <tr>
      <th>attempts</th>
      <td><p>Integer, number of attempts if server unresponsive, at default set to 2.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p>Boolean, whether to show messages of the job id for each batch. Default is FALSE.</p></td>
    </tr>
    <tr>
      <th>sequence</th>
      <td><p>A vector of strings representing protein amino acid sequences, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    <tr>
      <th>id</th>
      <td><p>A vector of strings representing protein identifiers, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p><a href='http://www.cbs.dtu.dk/services/SignalP/'>http://www.cbs.dtu.dk/services/SignalP/</a></p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>if org_type is "euk" a data frame with columns:</p><dl>
  <dt>id</dt><dd><p>Character, as from input</p></dd>
  <dt>Prediction</dt><dd><p>Integer, The type of signal peptide predicted: Sec/SPI, Tat/SPI, Sec/SPII or Other if no signal peptide predicted</p></dd>
  <dt>SP.Sec.SPI</dt><dd><p>Numeric, marginal probability that the protein contains a Sec  N-terminal signal peptide (Sec/SPI).</p></dd>
  <dt>Other</dt><dd><p>Numeric, the probability that the sequence does not have any kind of signal peptid.</p></dd>
  <dt>CS_pos</dt><dd><p>Character, cleavage site position.</p></dd>
  <dt>Pr</dt><dd><p>Numeric, probability of the predicted cleavage site position.</p></dd>
  <dt>cleave.site</dt><dd><p>Numeric,llocal amino acid sequence arround the predicted cleavage site.</p></dd>
  <dt>is.signalp</dt><dd><p>Logical, did SignalP5 predict the presence of a signal peptide.</p></dd>
  <dt>sp.length</dt><dd><p>Integer, length of the predicted signal peptide.</p></dd>
  
</dl>
  
<p>if org_type is one of "gram-", "gram+" or "archea" the returned data frame will have two additional columns between `SP.Sec.SPI` and `Other`:</p> 
<dl>
  <dt>TAT.Tat.SPI</dt><dd><p>Numeric, marginal probability that the protein contains a Tat N-terminal signal peptide (Tat/SPI).</p></dd>
  <dt>LIPO.Sec.SPII</dt><dd><p>Numeric, marginal probability that the protein contains a Lipoprotein N-terminal signal peptide (Sec/SPII).</p></dd>
  
</dl>

    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>This function creates temporary files in the working directory. If something goes wrong during communication with the server and progress was set to TRUE, predictions can be obtained using `file.path("http://www.cbs.dtu.dk/services/SignalP-5.0/tmp", jobid, "output_protein_type.txt")` eg `read.delim(file.path(...), header = TRUE, skip = 1)`.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Almagro Armenteros JJ, Tsirigos KD, Sønderby CK, Petersen TN, Winther O, Brunak S, von Heijne G, Nielsen H. (2019) SignalP 5.0 improves signal peptide predictions using deep neural networks. Nature Biotechnology, 37:420-423, doi:10.1038/s41587-019-0036-z</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='get_signalp.html'>get_signalp</a></code> <code><a href='get_targetp.html'>get_targetp</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/missuse/ragp'>ragp</a></span><span class='op'>)</span>
<span class='va'>sp5_pred</span> <span class='op'>&lt;-</span> <span class='fu'>get_signalp5</span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>at_nsp</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>10</span>,<span class='op'>]</span>,
                         <span class='va'>sequence</span>,
                         <span class='va'>Transcript.id</span><span class='op'>)</span>
<span class='va'>sp5_pred</span>
</div><div class='output co'>#&gt;             id  Prediction SP.Sec.SPI    Other CS_pos     Pr cleave.site
#&gt; 1  ATCG00660.1       OTHER   0.000375 0.999625            NA            
#&gt; 2  AT2G43600.1 SP(Sec/SPI)   0.999802 0.000198  22-23 0.9639      VFS-QN
#&gt; 3  AT2G28410.1 SP(Sec/SPI)   0.990424 0.009576  22-23 0.8897      ALA-QD
#&gt; 4  AT2G22960.1 SP(Sec/SPI)   0.998142 0.001858  22-23 0.9424      AES-GS
#&gt; 5  AT2G19580.1       OTHER   0.264792 0.735208            NA            
#&gt; 6  AT2G19690.2 SP(Sec/SPI)   0.989540 0.010460  28-29 0.9030      ARS-EE
#&gt; 7  AT2G19690.1 SP(Sec/SPI)   0.989540 0.010460  28-29 0.9030      ARS-EE
#&gt; 8  AT2G33130.1 SP(Sec/SPI)   0.959278 0.040722  26-27 0.5119      VVG-SR
#&gt; 9  AT2G05520.1 SP(Sec/SPI)   0.999493 0.000507  23-24 0.5666      VAA-AS
#&gt; 10 AT2G05520.2 SP(Sec/SPI)   0.999493 0.000507  23-24 0.5666      VAA-AS
#&gt;    is.signalp sp.length
#&gt; 1       FALSE        NA
#&gt; 2        TRUE        22
#&gt; 3        TRUE        22
#&gt; 4        TRUE        22
#&gt; 5       FALSE        NA
#&gt; 6        TRUE        28
#&gt; 7        TRUE        28
#&gt; 8        TRUE        26
#&gt; 9        TRUE        23
#&gt; 10       TRUE        23</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Milan Dragicevic, Danijela Paunovic.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


