<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>MAAB classification of hydroxyproline rich glycoproteins — maab • ragp</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="MAAB classification of hydroxyproline rich glycoproteins — maab"><meta property="og:description" content="Perform Motif and amino acid bias classification of hydroxyproline rich glycoproteins according to Johnson et al. (2017)"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ragp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.5.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pkgdown/start.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/pkgdown/filter.html">Filtering hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/analyse.html">Analyse hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/PSV.html">Protein Sequence Visualization</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://ragp.shinyapps.io/Rapp/" class="external-link">ragp shiny</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/missuse/ragp/" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MAAB classification of hydroxyproline rich glycoproteins</h1>
    <small class="dont-index">Source: <a href="https://github.com/missuse/ragp/blob/HEAD/R/maab.R" class="external-link"><code>R/maab.R</code></a></small>
    <div class="hidden name"><code>maab.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform Motif and amino acid bias classification of hydroxyproline rich glycoproteins according to Johnson et al. (2017)</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">maab</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for character</span>
<span class="fu">maab</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for data.frame</span>
<span class="fu">maab</span><span class="op">(</span><span class="va">data</span>, <span class="va">sequence</span>, <span class="va">id</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for list</span>
<span class="fu">maab</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for default</span>
<span class="fu">maab</span><span class="op">(</span>
  data <span class="op">=</span> <span class="cn">NULL</span>,
  <span class="va">sequence</span>,
  <span class="va">id</span>,
  order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ext"</span>, <span class="st">"tyr"</span>, <span class="st">"prp"</span>, <span class="st">"agp"</span><span class="op">)</span>,
  gpi <span class="op">=</span> <span class="cn">NULL</span>,
  get_gpi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bigpi"</span>, <span class="st">"predgpi"</span>, <span class="st">"netgpi"</span>, <span class="st">"none"</span><span class="op">)</span>,
  spec <span class="op">=</span> <span class="fl">0.99</span>,
  progress <span class="op">=</span> <span class="cn">FALSE</span>,
  <span class="va">...</span>
<span class="op">)</span>

<span class="co"># S3 method for AAStringSet</span>
<span class="fu">maab</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data</dt>
<dd><p>A data frame with protein amino acid sequences as strings in one column and corresponding id's in another. Alternatively a path to a .fasta file with protein sequences. Alternatively a list with elements of class <code><a href="https://rdrr.io/pkg/seqinr/man/SeqFastaAA.html" class="external-link">SeqFastaAA</a></code> resulting from <code><a href="https://rdrr.io/pkg/seqinr/man/read.fasta.html" class="external-link">read.fasta</a></code> call. Alternatively an <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">AAStringSet</a></code> object. Should be left blank if vectors are provided to sequence and id arguments.</p></dd>
<dt>...</dt>
<dd><p>currently no additional arguments are accepted apart the ones documented bellow.</p></dd>
<dt>sequence</dt>
<dd><p>A vector of strings representing protein amino acid sequences, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></dd>
<dt>id</dt>
<dd><p>A vector of strings representing protein identifiers, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path, or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></dd>
<dt>order</dt>
<dd><p>Order of motif counting, the default is as in Johnson et al. (2017).</p></dd>
<dt>gpi</dt>
<dd><p>A Boolean vector indicating if the corresponding id contains a GPI or not. Can be the 'is.bigpi' column from the output of get_big_pi.</p></dd>
<dt>get_gpi</dt>
<dd><p>A string indicating if <code><a href="get_big_pi.html">get_big_pi</a></code>,  <code><a href="get_pred_gpi.html">get_pred_gpi</a></code> or <code><a href="get_netGPI.html">get_netGPI</a></code> should be called on sequences that belong to one of the HRGP classes thus resolving class ambiguities that depend on GPI knowledge. At default set to 'none'.</p></dd>
<dt>spec</dt>
<dd><p>Numeric in the 0-1 range, indicating the threshold specificity of <code><a href="get_pred_gpi.html">get_pred_gpi</a></code>. Only valid if argument get_gpi = "predgpi".</p></dd>
<dt>progress</dt>
<dd><p>Boolean, whether to show the progress bar, at default set to FALSE.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data frame with columns:</p><ol><li><p>id protein identifiers as from input</p></li>
<li><p>ext_sp number of extensin SPn motifs, counted using SP3,5</p></li>
<li><p>ext_tyr number of extensin TYR motifs, sum of matches for: [FY].Y, KHY, VY[HKDE], V.Y, YY</p></li>
<li><p>prp number of proline rich protein motifs, sum of matches for: PPV.[KT], PPV[QK], KKPCPP</p></li>
<li><p>agp number of arabinogalactan motifs, sum of matches for: [AVTG]P1,3, [ASVTG]P1,2</p></li>
<li><p>past_percent summed percent of "P", "A", "S" and "T" amino acids</p></li>
<li><p>pvyk_percent summed percent of "P", "V", "Y" and "K" amino acids</p></li>
<li><p>psky_percent summed percent of "P", "S", "K" and "Y" amino acids</p></li>
<li><p>p_percent percent of "P"</p></li>
<li><p>coverage the coverage of sequence by the identified motifs</p></li>
<li><p>maab_class determined maab class</p></li>
</ol></div>
    <div id="details">
    <h2>Details</h2>
    <p>The function provides motif and amino acid bias descriptors used for classification of HRGP's by the MAAB pipeline (Johnson et al. 2017) as well as the determined HRGP classes. The motifs are counted in a specific order ext &gt; tyr &gt; prp &gt; agp, and overlapping motifs are not counted. Hence the classification depends on the order of counting, this is most noticeable for tyr and prp, we recommend using both the default order and  'order = c("ext", "prp","tyr", "agp")'.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Johnson KL, Cassin AM, Lonsdale A, Bacic A, Doblin MS, Schultz CJ. (2017) Pipeline to Identify Hydroxyproline-Rich Glycoproteins. Plant Physiol 174(2): 886-903.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="scan_ag.html">scan_ag</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/missuse/ragp" class="external-link">ragp</a></span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">at_nsp</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">maab_class</span> <span class="op">&lt;-</span> <span class="fu">maab</span><span class="op">(</span>sequence <span class="op">=</span> <span class="va">at_nsp</span><span class="op">$</span><span class="va">sequence</span>,</span>
<span class="r-in">                   id <span class="op">=</span> <span class="va">at_nsp</span><span class="op">$</span><span class="va">Transcript.id</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">maab_class</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            id ext_sp ext_tyr prp agp past_percent pvyk_percent psky_percent</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 ATCG00660.1      0       1   0   0     18.80342     14.52991     17.94872</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 AT2G43600.1      0       6   0   4     21.61172     23.44322     26.73993</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 AT2G28410.1      0       0   0   4     37.39130     18.26087     22.60870</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 AT2G22960.1      0       0   0   3     23.91304     17.93478     23.36957</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 AT2G19580.1      0       1   0   2     22.96296     21.85185     18.88889</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 AT2G19690.2      0       0   0   0     20.94595     20.94595     19.59459</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   p_percent   coverage maab_class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  0.000000 0.02564103          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  3.663004 0.08791209          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  7.826087 0.06956522          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  5.434783 0.03260870          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  4.074074 0.02592593          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  2.702703 0.00000000          0</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Milan Dragicevic, Danijela Paunovic.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

