<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Detection of motifs for N-glycosylation on asparagine residues. — scan_nglc • ragp</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Detection of motifs for N-glycosylation on asparagine residues. — scan_nglc"><meta property="og:description" content="Detection is based on PROSITE pattern PS00001. Mean local hydrophilicity (Hopp and Woods, 1981) is used to assess if the asparagines are buried."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ragp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.5.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pkgdown/start.html">Getting started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/pkgdown/filter.html">Filtering hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/analyse.html">Analyse hydroxyproline rich glycoproteins</a>
    </li>
    <li>
      <a href="../articles/pkgdown/PSV.html">Protein Sequence Visualization</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://ragp.shinyapps.io/Rapp/" class="external-link">ragp shiny</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/missuse/ragp/" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Detection of motifs for N-glycosylation on asparagine residues.</h1>
    <small class="dont-index">Source: <a href="https://github.com/missuse/ragp/blob/HEAD/R/scan_nglc.R" class="external-link"><code>R/scan_nglc.R</code></a></small>
    <div class="hidden name"><code>scan_nglc.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Detection is based on PROSITE pattern PS00001. Mean local hydrophilicity (Hopp and Woods, 1981) is used to assess if the asparagines are buried.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">scan_nglc</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for character</span>
<span class="fu">scan_nglc</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for data.frame</span>
<span class="fu">scan_nglc</span><span class="op">(</span><span class="va">data</span>, <span class="va">sequence</span>, <span class="va">id</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for list</span>
<span class="fu">scan_nglc</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for default</span>
<span class="fu">scan_nglc</span><span class="op">(</span>data <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">sequence</span>, <span class="va">id</span>, span <span class="op">=</span> <span class="fl">5L</span>, cutoff <span class="op">=</span> <span class="fl">0</span>, nsp <span class="op">=</span> <span class="fl">15L</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S3 method for AAStringSet</span>
<span class="fu">scan_nglc</span><span class="op">(</span><span class="va">data</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://prosite.expasy.org/PDOC00001" class="external-link">https://prosite.expasy.org/PDOC00001</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data</dt>
<dd><p>A data frame with protein amino acid sequences as strings in one column and corresponding id's in another. Alternatively a path to a .fasta file with protein sequences. Alternatively a list with elements of class <code><a href="https://rdrr.io/pkg/seqinr/man/SeqFastaAA.html" class="external-link">SeqFastaAA</a></code> resulting from <code><a href="https://rdrr.io/pkg/seqinr/man/read.fasta.html" class="external-link">read.fasta</a></code> call. Alternatively an <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">AAStringSet</a></code> object. Should be left blank if vectors are provided to sequence and id arguments.</p></dd>
<dt>...</dt>
<dd><p>currently no additional arguments are accepted apart the ones documented bellow.</p></dd>
<dt>sequence</dt>
<dd><p>A vector of strings representing protein amino acid sequences, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></dd>
<dt>id</dt>
<dd><p>A vector of strings representing protein identifiers, or the appropriate column name if a data.frame is supplied to data argument. If .fasta file path or list with elements of class "SeqFastaAA" provided to data, this should be left blank.</p></dd>
<dt>span</dt>
<dd><p>An integer specifying how many amino acids around the target asparagine residues is used to calculate hydrophilicity. At default set to 5: asparagine position - 5 to asparagine position +5 residues. Range to consider: 3 - 10. Acceptable values are 0 - 20.</p></dd>
<dt>cutoff</dt>
<dd><p>An integer specifying the cutoff value for hydrophilicity. Range to consider: -1 - 1. Values lower then -3.4 exclude hydrophilicity as a parameter, while values higher than 3 result in no motifs being found.</p></dd>
<dt>nsp</dt>
<dd><p>An integer, either of length 1 or length equal the number of sequences, specifying the number of N-terminal amino acids to exclude.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data frame with columns:</p><dl><dt>id</dt>
<dd><p>Character, as supplied in the function call.</p></dd>

  <dt>align_start</dt>
<dd><p>Integer, start of motif match.</p></dd>

  <dt>motif</dt>
<dd><p>Character, the motif matched.</p></dd>

  <dt>hydrophilicity</dt>
<dd><p>the average hydrophilicity.</p></dd>

  <dt>is.nglc</dt>
<dd><p>Boolean, is the N-glycosylation likely based on hydrophilicity.</p></dd>

  <dt>nsp</dt>
<dd><p>Optional integer column provided when nsp argument is of equal length to the number of input sequences</p></dd>


</dl></div>
    <div id="note">
    <h2>Note</h2>
    <p>For N-glycosylation to happen the protein must enter the endoplasmic reticulum. Please check if the proteins are likely to contain a N-terminal signal peptide. The motif Asp-Xaa-Ser/Thr (where Xaa is not Pro) on which N-glycosylation occurs is relatively common, however for N-glycosylation to occur the motif needs to be located on the protein surface. Mean local hydrophilicity (Hopp and Woods, 1981) is used here to evaluate if the asparagines are located in a hydrophilic surrounding which is more likely on the protein surface.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Hopp TP. Woods KR. (1981) Prediction of protein antigenic determinants from amino acid sequences. Proceedings of the National Academy of Sciences of the United States of America, 78(6): 3824-8</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>original R code by Thomas Shafee, modified by Milan Dragićević</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/missuse/ragp" class="external-link">ragp</a></span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">at_nsp</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">nglc_pred</span> <span class="op">&lt;-</span> <span class="fu">scan_nglc</span><span class="op">(</span>data <span class="op">=</span> <span class="va">at_nsp</span>,</span>
<span class="r-in">                       sequence <span class="op">=</span> <span class="va">sequence</span>,</span>
<span class="r-in">                       id <span class="op">=</span> <span class="va">Transcript.id</span><span class="op">)</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Milan Dragicevic, Danijela Paunovic.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

